log.go

```diff
@@ -41,6 +41,9 @@ type NewCPFunc func(size uint64, hash []byte) ([]byte, error)
 // ParseCPFunc is the signature of a function which knows how to verify and parse checkpoints.
 type ParseCPFunc func(raw []byte) (*f_log.Checkpoint, error)
 
+// IndexFuture is the signature of a function which can return an assigned index or error.
```

Comment:
Obvious to people that know about futures, but worth adding a comment to mention that this will block?

Reply:
Yeah, good call - done.